docs/docs/preview/iframebridge/IFrameBridge.md

```diff
@@ -0,0 +1,65 @@
+| `IFrameHoverComponentMessage`  | Site   | The `IFrameHoverComponentMessage` is sent from the site to the admin, when a component is hovered in the iFrame and should be highlighted in the admin interface.                                             | `iFrameBridge.sendHoverComponent("route")` |
+| `IFrameOpenLinkMessage`        | Site   | @deprecated                                                                                                                                                                                                   |
+| `IFrameLocationMessage`        | Site   | @deprecated                                                                                                                                                                                                   |
+| `IAdminBlockMessage`           | Admin  | The `IAdminBlockMessage` is sent from the admin to the site, which block should be displayed in the iFrame.                                                                                                   | `iFrameBridge.sendBlockState()`            |
+| `IAdminShowOnlyVisibleMessage` | Admin  | The `IAdminBlockMessage` is sent from the admin to the site, which block should be displayed in the iFrame.                                                                                                   | `iFrameBridge.sendShowOnlyVisible(true)`   |
+| `IAdminSelectComponentMessage` | Admin  | The `IAdminSelectComponentMessage` is sent from the admin to the site, when a component is selected in the admin interface.                                                                                   | `iFrameBridge.sendSelectComponent("id")`   |
+| `IAdminHoverComponentMessage`  | Admin  | The `IAdminHoverComponentMessage` is sent from the admin to the site, when a component is hovered in the admin interface.                                                                                     | `iFrameBridge.sendHoverComponent("id")`    |
+| `IAdminContentScopeMessage`    | Admin  | The `IAdminContentScopeMessage` is sent from the admin to the site, when the content scope is changed in the admin interface.                                                                                 | `iFrameBridge.sendContentScope(scope)`     |
+
+## block-preview
+
+To make the Block Preview embeddable in the Comet Admin, the client must provide an entry point where the client application renders the block preview and implements the `IFrameBridge` for communication. In a typical comet application the block preview is available at `site-domain.com/block-preview/page`
+
+To Implement the `IFrameBridge` in any site client, you can use `IFrameBridgeProvider` and `useIFrameBridge` from `@comet/cms-site`.
```

Comment:
I believe you'd use the higher level `BlockPreviewProvider`

Reply:
I was not aware of the BlockPreviewProvider, but i don't think that the BlockPreviewProvider is here the way to go. 

The only thing BlockPreviewProvider is doing, is storing a state of the previewType and if Preview Skeletons should be shown.

<img width="1008" alt="Screenshot 2024-11-29 at 08 50 00" src="https://github.com/user-attachments/assets/a5dac0c4-2ec0-4d82-908a-896197eac416">

If you want to communicate between admin and site, one has to use the IFrameBridge.